client/src/features/session/components/options/SessionUserSecrets.tsx

```diff
@@ -0,0 +1,239 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+import cx from "classnames";
+import { useCallback, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { Button, Card, CardBody, Collapse, Input, Label } from "reactstrap";
+
+import { Loader } from "../../../../components/Loader";
+import { User } from "../../../../model/renkuModels.types";
+import useLegacySelector from "../../../../utils/customHooks/useLegacySelector.hook";
+import { Url } from "../../../../utils/helpers/url";
+import ChevronFlippedIcon from "../../../../components/icons/ChevronFlippedIcon";
+import { useGetSecretsQuery } from "../../../secrets/secrets.api";
+import { RtkOrNotebooksError } from "../../../../components/errors/RtkErrorAlert";
+import { setSecretsList, setSecretsPath } from "../../startSessionOptionsSlice";
+import useAppDispatch from "../../../../utils/customHooks/useAppDispatch.hook";
+import useAppSelector from "../../../../utils/customHooks/useAppSelector.hook";
+import { SessionSecrets } from "../../startSessionOptions.types";
+
+export default function SessionUserSecrets() {
+  const secretsUrl = Url.get(Url.pages.secrets);
+  const user = useLegacySelector<User>((state) => state.stateModel.user);
+
+  if (!user.fetched) return <Loader />;
+
+  const content = user.logged ? (
+    <>
+      <div className={cx("form-text", "my-1")}>
+        You can select secrets defined in the{" "}
+        <Link to={secretsUrl}> User Secrets page</Link> and mount them as files
+        in your session.
+      </div>
+      <SessionUserSecretsSection />
+    </>
+  ) : (
+    <div className={cx("form-text", "my-1")}>
+      This feature is only available to logged users.
+    </div>
+  );
+
+  return (
+    <div className="field-group" data-cy="session-secrets">
+      <div className="form-label">User Secrets</div>
+      {content}
+    </div>
+  );
+}
+
+function SessionUserSecretsSection() {
+  // Handle the collapse
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleIsOpen = useCallback(() => setIsOpen((isOpen) => !isOpen), []);
+
+  // Fetch the secrets
+  const secrets = useGetSecretsQuery();
+
+  // Get current values from the store
+  const sessionOptions = useAppSelector((state) => state.startSessionOptions);
+
+  // Save data in the Redux store
+  const dispatch = useAppDispatch();
+  const changeSecretsPath = useCallback(
+    (newValue: string) => {
+      dispatch(setSecretsPath(newValue));
+    },
+    [dispatch]
+  );
+  const updateSecretsList = useCallback(
+    (secret: SessionSecrets) => {
+      if (sessionOptions.secretsList.map((s) => s.name).includes(secret.name)) {
+        dispatch(
+          setSecretsList(
+            sessionOptions.secretsList.filter((s) => s.name !== secret.name)
+          )
+        );
+      } else {
+        dispatch(setSecretsList([...sessionOptions.secretsList, secret]));
+      }
```

Comment:
See the  proposal for a `toggleSecret` action in the slice file
```suggestion
        dispatch(toggleSecret(secret));
        }
```